configuration/reports/reportdescriptors/dataexports/sql/allergies.sql

```diff
@@ -0,0 +1,35 @@
+CALL initialize_global_metadata();
+
+SELECT p.patient_id, dos.identifier dossierId, zl.identifier zlemr, zl_loc.name loc_registered, a.date_created,
+CONCAT(pn.given_name, ' ',pn.family_name) "creator",
+a.date_changed,
+CONCAT(pn_ch.given_name, ' ',pn_ch.family_name) "changed_by",
+a.allergen_type,
+cn_allergen.name 'allergen',
+cn_severity.name 'severity',
+group_concat(cn_reaction.name separator ',') 'reaction',
+a.comments "comments"
+FROM patient p
+INNER JOIN allergy a on a.patient_id = p.patient_id and a.voided = 0
+-- Most recent Dossier ID
+LEFT OUTER JOIN (SELECT patient_id, identifier, location_id FROM patient_identifier WHERE identifier_type =@dosId
```

Comment:
Note that I changed all reports to the new "SqlFile" definition type, so everything like ":dosId" needed to be replaced with "@dosId" and the appropriate variable needs to be defined in the "initialized_global_metadata()" procedure